cmd/webhook-tester/main.go

```diff
@@ -20,193 +21,318 @@ import (
-		fmt.Printf("--- Running test case: %s ---\n", tc.name)
-
-		signature, err := tc.signer([]byte(tc.payload), *secret)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error generating signature for test case %q: %v\n", tc.name, err)
-			os.Exit(1)
+	if *installationID == 0 {
+		return fmt.Errorf("--installation-id is required")
+	}
+	if *projectID == "" {
+		return fmt.Errorf("--project-id is required")
+	}
+	if *runID == "" {
+		return fmt.Errorf("--run-id is required")
+	}
+	if *verifyRunner {
+		if *githubOwner == "" {
+			return fmt.Errorf("--github-owner is required with --verify-runner")
 		}
-
-		ctx := context.Background()
-		req, err := http.NewRequestWithContext(ctx, "POST", *targetURL, bytes.NewReader([]byte(tc.payload)))
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error creating request for test case %q: %v\n", tc.name, err)
-			os.Exit(1)
+		if *githubRepo == "" {
+			return fmt.Errorf("--github-repo is required with --verify-runner")
 		}
+	}
 
-		req.Header.Set("Content-Type", "application/json")
-		req.Header.Set("X-GitHub-Event", tc.eventHeader)
-		if signature != "" {
-			req.Header.Set("X-Hub-Signature-256", signature)
-		}
+	if *payload == "" && *payloadFile == "" {
+		return fmt.Errorf("either --payload or --payload-file is required")
+	}
 
-		client := &http.Client{Timeout: 15 * time.Second}
-		resp, err := client.Do(req)
+	secret, err := getSecret(ctx, *secretName)
+	if err != nil {
+		return fmt.Errorf("failed to get secret: %w", err)
+	}
+
+	body := *payload
+	if *payloadFile != "" {
+		payloadBytes, err := os.ReadFile(*payloadFile)
 		if err != nil {
-			fmt.Fprintf(os.Stderr, "Error sending request for test case %q: %v\n", tc.name, err)
-			os.Exit(1)
+			return fmt.Errorf("failed to read payload file: %w", err)
 		}
+		body = string(payloadBytes)
+	}
 
-		body, readErr := io.ReadAll(resp.Body)
-		if readErr != nil {
-			fmt.Fprintf(os.Stderr, "Error reading response body for test case %q: %v\n", tc.name, readErr)
-			resp.Body.Close() // Close the body before exiting
-			os.Exit(1)
-		}
-		resp.Body.Close() // Ensure the body is closed after reading
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, *webhookURL, bytes.NewBufferString(body))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	sig := *signatureFlag
+	if sig == "" {
+		sig = signature([]byte(secret), []byte(body))
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("X-GitHub-Event", "workflow_job")
+	req.Header.Set("X-Hub-Signature-256", "sha256="+sig)
+	if *idToken != "" {
+		req.Header.Set("Authorization", "Bearer "+*idToken)
+	}
 
-		fmt.Printf("Response Status: %s\n", resp.Status)
-		fmt.Printf("Response Body: %s\n", string(body))
+	client := &http.Client{
+		Timeout: 30 * time.Second,
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	respBody, err := io.ReadAll(resp.Body)
```

Comment:
I know this is for testing, but you probably want to use a LimitReader with some sane max size just in case.

Something like:

`respBody, err := io.LimitReader(resp.Body, 4_194_304)`